lib/teiserver/player/session.ex

```diff
@@ -964,6 +972,12 @@ defmodule Teiserver.Player.Session do
     {:noreply, state}
   end
 
+  def handle_cast({:user, {:role_updated, roles}}, state) do
+    send_to_player!({:user, {:role_updated, roles}}, state)
+    new_state = %{state | user: %{state.user | roles: roles}}
```

Comment:
This is fine, but there's an easier way on doing that: `put_in(state.user.roles, roles)` see [put_in](https://hexdocs.pm/elixir/Kernel.html#put_in/2)